src/actions/walletActions.js

```diff
@@ -155,71 +159,138 @@ export function autoSelectCoins() {
+      );
     }
 
-    // check the next deposit slice just in case a spend is to own wallet
-    // this doesn't catch all self-spend cases but should catch the majority
-    // to avoid any confusion for less technical users.
-    sliceUpdates.push(
-      fetchSliceStatus(nextNode.multisig.address, nextNode.bip32Path)
-    );
+    // once all queries have completed we can confirm which have been updated
+    // and dispatch the changes to the store
+    for (let i = 0; i < queriedSlices.length; i += 1) {
+      const slice = queriedSlices[i];
+      const utxoCount = slice.change
+        ? changeSlices[slice.bip32Path].utxos.length
+        : depositSlices[slice.bip32Path].utxos.length;
 
-    const updatedSlices = await Promise.all(sliceUpdates);
-
-    updatedSlices.forEach((slice) => {
-      if (slice.change)
-        dispatch({
-          type: UPDATE_CHANGE_SLICE,
-          value: slice,
-        });
-      else
+      // if the utxo count is not the same then we can reliably update
+      // this slice and can skip in any future calls
+      if (slice && slice.utxos && slice.utxos.length !== utxoCount) {
         dispatch({
-          type: UPDATE_DEPOSIT_SLICE,
+          type: slice.change ? UPDATE_CHANGE_SLICE : UPDATE_DEPOSIT_SLICE,
           value: slice,
         });
-    });
+        skipAddresses.add(slice.address);
+      }
+    }
+
+    // if not all slices queried were successful, then we want to recursively call
+    // updateTxSlices, counting down retries and with the full set of successful queries
+    // ALL input slices must return a different utxo set otherwise something went wrong
+    if (skipAddresses.size !== queriedSlices.length && retries)
+      return setTimeout(
+        () => dispatch(updateTxSlices(changeSlice, retries - 1, skipAddresses)),
+        750
+      );
+
+    // if we're out of retries and counts are still the same
+    // then we're done trying and should show an error
+    if (!retries) {
+      let message = `There was a problem updating the wallet balance.
+      It is recommended you refresh the wallet to make sure UTXO current set is up to date.`;
+      if (txid && txid.length) message += ` Transaction ID: ${txid}`;
+      return dispatch(setErrorNotification(message));
+    }
+
+    // Check the next deposit slice just in case a spend is to own wallet.
+    // This doesn't catch all self-spend cases but should catch the majority
+    // to avoid any confusion for less technical users.
+    const updatedSlice = await fetchSliceStatus(
+      nextDepositSlice.multisig.address,
+      nextDepositSlice.bip32Path
+    );
+
+    // if its status has changed and the utxo set for the next
+    // deposit slice is different, then update it as well
+    if (
+      updatedSlice.utxos.length !==
+      depositSlices[updatedSlice.bip32Path].utxos.length
+    )
+      dispatch({ type: UPDATE_DEPOSIT_SLICE, value: updatedSlice });
+
+    dispatch(setErrorNotification("Wallet addresses updated"));
```

Comment:
What does this mean exactly?

Reply:
Yeah, I wasn't sure how best to express this in a user friendly way. The actual result if we've gotten this far is that the state of all input utxos and change if there was one has been updated according to the blockchain client, and the next deposit address has been checked and possibly updated as well. I'm open to other suggestions, maybe "Local Address state updated"? Also we can remove it and just leave as "No news is good news", i.e. no error message popped up, so should be good.

Reply:
I tend to be on the "no news is good news" side of things.  Maybe we can just remove this.